src/org/daisy/dotify/formatter/impl/obfl/ObflParserImpl.java

```diff
@@ -116,6 +116,7 @@
     Map<String, Node> xslts = new HashMap<>();
     Map<String, Node> fileRefs = new HashMap<>();
     Map<String, List<RendererInfo>> renderers = new HashMap<>();
+    private HashMap<QName, String> externalReferenceObject = null;
```

Comment:
why is this map called 'externalReferenceObject' and not 'externalReferenceObjects`?
and what does it map? - this is not clear from the declaration.  Please use a clear name or add a comment that indicates what is being mapped here.

Reply:
This is the externalReferenceObject that could be anything but in this specific implementation seems to be a map. In order to check when it hasn't been initialized we use null as we don't want any extra data sent if not used.

Reply:
Maybe I'm just too critical. But I will try to explain in more detail what I don't like about this statement.

1. It is unclear what is being mapped here. It maps a QName to a String. What does the String represent? That is not clear from the name 'externalReferenceObject` and therefore there should be a comment that describes what is being mapped, especially, what the target of the mapping is.
2. Usually when you declare a map, you abstract from the implementation. That is, normally you would have something like `private Map<...> nameOfMap = ...` instead of `private HashMap<...> = ...`. Please have a look at all the code in this repository and you will see that a map is always _declared_ as a Map, although it may be _implemented_ as a HashMap. A solution would be to have here: `private HashMap<QName, String> externalReferenceObject = (HashMap)null`. But that looks fishy to me, see the next point.
3. I think it is a glitch to initialize a map with `null`, in general. This should be avoided if possible.

Reply:
From the name "externalReferenceObject" it is pretty obvious to me that it is about the "external-reference" OBFL element, no? So what else can the map be than a set of attribute name/value pairs? I think the name is fine.

I agree with not using HashMap as the variable type, but I'm not sure that you will not find any other examples of this in this repository.